lib/ex_doc/retriever.ex

```diff
@@ -158,16 +158,18 @@ defmodule ExDoc.Retriever do
   defp generate_node(module, type, config) do
     source_url  = config.source_url_pattern
     source_path = source_path(module, config)
+    config = %{source_url: source_url, source_path: source_path}
 
     specs = get_specs(module)
     impls = get_impls(module)
     abst_code = get_abstract_code(module)
+    module_parts = %{specs: specs, impls: impls, abst_code: abst_code}
```

Comment:
Let's call it `module_info` and include the module name (`module`) and module type (`type`) on it? Then we can move its creation into something called `get_module_info(module)`.